Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2113,7 +2113,13 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             let copyAction = UIAlertAction(title: copyImageTitle, style: UIAlertActionStyle.Default) { (action: UIAlertAction) -> Void in
                 let pasteBoard = UIPasteboard.generalPasteboard()
                 pasteBoard.string = url.absoluteString
-                // TODO: put the actual image on the clipboard
+                // put the actual image on the clipboard
+                // do this be asyncronously just in case we're in a low bandwidth situation
+                // Is this neccessary, or is the image already sat in cache?
+                NSURLSession.sharedSession().dataTaskWithURL(url, completionHandler: { (data, _, error) -> Void in
```

Comment:
Not sure - doesn't seem there is an optimal solution given the constraint of the private API. I see three non-JS options:

1. Silently download the image and in cases that the image is large or the network is slow, the image doesn't get copied

2. Block the user with a progress bar until the image is copied

3. Add a loading indicator to inform the user the image is downloading in the background with a toast of some sort to indicate completion.

I don't really like any of these options. I think the biggest problem is that copying an image should imply that it's an instant operation which doesn't require a network request and showing the indicator (blocking or non-blocking) removes that illusion. Although a blocking indicator would be intrusive, a non-blocking one would require us to find a place to put it. In the case of failure with option 1, it make it look like the app has a bug and the feature doesn't work. If I had to choose one I'd probably stick with what we have and go with option 1 unless there is a better way in JS.